src/main/java/com/powsybl/openloadflow/dc/AbstractWoodburyEngineInputReader.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.dc;
+
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.openloadflow.network.LfBus;
+import com.powsybl.openloadflow.network.LfHvdc;
+import com.powsybl.openloadflow.network.impl.Networks;
+import com.powsybl.openloadflow.network.util.ParticipatingElement;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
+ * @author Gaël Macherel {@literal <gael.macherel@artelys.com>}
+ */
+public abstract class AbstractWoodburyEngineInputReader implements WoodburyEngineInputReader {
+
+    /**
+     * @return True if the disabled buses change the slack distribution.
+     */
+    public boolean hasRhsChangedDueToDisabledSlackBus(LoadFlowParameters lfParameters, Set<LfBus> disabledBuses, List<ParticipatingElement> participatingElements) {
+        return lfParameters.isDistributedSlack() && participatingElements.stream().anyMatch(element -> disabledBuses.contains(element.getLfBus()));
```

Comment:
Why is the method public ? Isn't protected sufficient ?

Reply:
Yes you are right.